src/netconf_server.c

```diff
@@ -32,517 +35,153 @@
 {
-    sr_change_iter_t *iter;
-    sr_change_oper_t op;
-    const struct lyd_node *node;
-    int rc, idle_time = -1, max_probes = -1, probe_interval = -1;
+    /* use POSIX username definition
+     * https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_437 */
 
-    rc = sr_get_changes_iter(session, xpath, &iter);
-    if (rc != SR_ERR_OK) {
-        ERR("Getting changes iter failed (%s).", sr_strerror(rc));
-        return rc;
+    /* not empty */
+    if (strlen(username) == 0) {
+        return -1;
     }
 
-    while ((rc = sr_get_change_tree_next(session, iter, &op, &node, NULL, NULL, NULL)) == SR_ERR_OK) {
-        if (!strcmp(node->schema->name, "idle-time")) {
-            if (op == SR_OP_DELETED) {
-                idle_time = 1;
-            } else {
-                idle_time = ((struct lyd_node_term *)node)->value.uint16;
-            }
-        } else if (!strcmp(node->schema->name, "max-probes")) {
-            if (op == SR_OP_DELETED) {
-                max_probes = 10;
-            } else {
-                max_probes = ((struct lyd_node_term *)node)->value.uint16;
-            }
-        } else if (!strcmp(node->schema->name, "probe-interval")) {
-            if (op == SR_OP_DELETED) {
-                probe_interval = 5;
-            } else {
-                probe_interval = ((struct lyd_node_term *)node)->value.uint16;
-            }
-        }
-    }
-    sr_free_change_iter(iter);
-    if (rc != SR_ERR_NOT_FOUND) {
-        ERR("Getting next change failed (%s).", sr_strerror(rc));
-        return rc;
+    /* no hyphen as first char */
+    if (username[0] == '-') {
+        return -1;
     }
 
-    rc = 0;
-
-    /* set new keepalive parameters */
-    if (!client_name) {
-        if (nc_server_is_endpt(endpt_name)) {
-            rc = nc_server_endpt_set_keepalives(endpt_name, idle_time, max_probes, probe_interval);
-        }
-    } else {
-        if (nc_server_ch_client_is_endpt(client_name, endpt_name)) {
-            rc = nc_server_ch_client_endpt_set_keepalives(client_name, endpt_name, idle_time, max_probes, probe_interval);
+    /* check for Portable Filename Character Set */
+    for (unsigned long i = 0; i < strlen(username); i++) {
+        if (!(isalnum(username[i]) || (username[i] == '.') || (username[i] == '_') || (username[i] == '-'))) {
+            return -1;
         }
     }
-    if (rc) {
-        ERR("Keepalives configuration failed (%d).", rc);
-        return SR_ERR_INTERNAL;
-    }
 
-    return SR_ERR_OK;
+    return 0;
 }
 
-/* /ietf-netconf-server:netconf-server/listen/endpoint/ * /tcp-server-parameters */
 int
-np2srv_endpt_tcp_params_cb(sr_session_ctx_t *session, uint32_t UNUSED(sub_id), const char *UNUSED(module_name),
-        const char *xpath, sr_event_t UNUSED(event), uint32_t UNUSED(request_id), void *UNUSED(private_data))
+np2srv_pubkey_auth_cb(const struct nc_session *session, ssh_key key, void *UNUSED(user_data))
```

Comment:
Server (libnetconf2) should use the authorized keys configured for the client, not read them from the system or am I missing something?